### PR TITLE
Update nixpkgs, static-haskell-nix and use GHC 8.10.4

### PR DESCRIPTION
**Pull request by monacoremo**
Time to upgrade nixpkgs again. @nh2 did amazing work that makes this possible, thanks a million! 

* [x] Upgrade `nixpkgs` to a version that includes https://github.com/NixOS/nixpkgs/pull/129289`
* [x] Fix linting errors from new hlint version
* [x] Fix io-tests after change in the `jwt` Python module API
* [x] Revive Postgres 9.5 as it dropped out of support and was removed from nixpkgs
* [x] Upgrade `static-haskell-nix`, leveraging https://github.com/nh2/static-haskell-nix/pull/98
* [x] Set our Nix setup to use GHC 8.10.4 (`hsie` to follow later due to changes in the GHC API)
* [x] Upgrade `stackage` to LTS 18.2 with GHC 8.10.4

### CODE REVIEW

**Comment by nh2** [comment]
Let me know if you encounter any problems.

**Comment by monacoremo** [comment]
> Let me know if you encounter any problems.

Thank you @nh2! Seems to work perfectly. Only slight hiccup: The statically built executable was no longer stripped by default, which was easily fixed with a `lib.doStrip`. Are you aware whether that was an intended change somewhere in the Nix Haskell packages or static-haskell-nix?

File: nix/overlays/postgresql-legacy.nix [hunk 1]
```diff
@@ -0,0 +1,20 @@
+self: super:
+# Overlay that adds legacy versions of PostgreSQL that are supported by
+# PostgREST.
+{
+  # PostgreSQL 9.5 was removed from Nixpkgs with
+  # https://github.com/NixOS/nixpkgs/commit/72ab382fb6b729b0d654f2c03f5eb25b39f11fbb
+  # We pin its parent commit to get the last version that was available.
+  postgresql_9_5 =
```
Comment (wolfgangwalther) [inline nix/overlays/postgresql-legacy.nix:8]: Not sure if we need that - we agreed in #1820 to remove 9.5 soon anyway.

This boils down to whether we'll release v8 before merging this here or after.

@steve-chavez WDYT? How close are we do getting v8 ready docs-wise?

File: test/io-tests/test_io.py [hunk 2]
```diff
@@ -207,7 +207,7 @@ def authheader(token):
 
 def jwtauthheader(claim, secret):
     "Authorization header with signed JWT."
-    return authheader(jwt.encode(claim, secret).decode("utf-8"))
+    return authheader(jwt.encode(claim, secret))
```
Comment (wolfgangwalther) [inline test/io-tests/test_io.py:210]: Is this change related to the other changes?

**Review by wolfgangwalther — APPROVED** [review]
I didn't check the changes to the patches for other nix packages - those are over my head ;).

File: test/io-tests/test_io.py (hunk 2 above)
Comment (monacoremo) [reply test/io-tests/test_io.py:210]: Yes, the API of the Python jwt module changed with the newer version in the updated nixpkgs, returning a string instead of bytes: https://pyjwt.readthedocs.io/en/latest/changelog.html#jwt-encode-return-type

File: nix/overlays/postgresql-legacy.nix (hunk 1 above)
Comment (monacoremo) [reply nix/overlays/postgresql-legacy.nix:8]: Suggest that we merge this before, as we'll bake the latest 13.3 Postgres bindings into the static executable that way (not that there is anything critical there, but can't hurt: https://www.postgresql.org/docs/release/13.3/).

In any case, I would also keep this file around, with commented out content if it's not needed right now - we'll need it now and again later.

Comment (steve-chavez) [reply nix/overlays/postgresql-legacy.nix:8]: Hey guys, good to see you back!

> This boils down to whether we'll release v8 before merging this here or after.

I don't see a problem with removing 9.5 after v8, so we can merge this.

> How close are we do getting v8 ready docs-wise?

@wolfgangwalther There's still work to be done in the docs. Me and @laurenceisla thought it'd be better to tag/release v8 with just the CHANGELOG in the github release(next week tops, after finishing the last breaking change https://github.com/PostgREST/postgrest/issues/1643) - and once that's done we can take a bit more time with finishing the v8 docs.

**Review by steve-chavez — APPROVED** [review]

**Comment by nh2** [comment]
> The statically built executable was no longer stripped by default, which was easily fixed with a `lib.doStrip`. Are you aware whether that was an intended change somewhere in the Nix Haskell packages or static-haskell-nix?

@monacoremo Yes, that is this issue that made `strip` 300x on `.a` files:

https://github.com/nh2/static-haskell-nix/pull/98/commits/da452685fc7c5742c45cec3386631782884fa006

So I enabled `dontStrip` for all `.a` files to not make the build take insanely long. Your approach to strip your final executable as desired is correct.

I have already landed a fix for the 300x slowdown on nixpkgs `staging`, but I'm considering to keep the `dontStrip = true;` the default, so that people can decide at the very end whether they prefer small executable sizes, increased debuggability by keeping debug symbols. Still on the fence which way to go, as I haven't tried GHC's DWARF support for a while.

**Comment by nh2** [comment]
@monacoremo I've pushed another update that should make `doStrip` unneccessary:

https://github.com/nh2/static-haskell-nix/pull/98/commits/cdc1a9a36add24fac3b98e32b03e0e3c41bd17ee

But your approach here is also fine.

**Comment by monacoremo** [comment]
> @monacoremo I've pushed another update that should make `doStrip` unneccessary:
> 
> [nh2/static-haskell-nix@cdc1a9a](https://github.com/nh2/static-haskell-nix/commit/cdc1a9a36add24fac3b98e32b03e0e3c41bd17ee)

Your fix worked perfectly @nh2, many thanks! Again, amazing work on fixing musl GHC with all those interlinked issues that cropped up.
